PSix/ParseUserSession.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  CurrentUser.swift
+//  PSix
+//
+//  Created by Turzion, Avihu on 4/14/15.
+//  Copyright (c) 2015 PayPal. All rights reserved.
+//
+
+import Foundation
+import Parse
+
+class ParseUserSession {
+    
+    private static let FB_READ_PERMISSIONS = ["public_profile", "user_events"]
+    
+    static private var currentLoggedInUser: User? = nil
+    
+    static var currentUser: User? {
+        get {
+            if currentLoggedInUser == nil && PFUser.currentUser() != nil {
+                currentLoggedInUser = User(withoutDataWithObjectId: PFUser.currentUser()?.objectId)
+                currentLoggedInUser?.fetch()
+            } else if currentLoggedInUser == nil && PFUser.currentUser() == nil {
+                return nil
+            }
+            return currentLoggedInUser
+        }
```

Comment:
Okay, I've fixed the issue with subclassing `PFUser` and now this code looks significantly cleaner.